cognite_toolkit/_cdf_tk/client/api/extended_data_modeling.py

```diff
@@ -8,3 +27,170 @@ class ExtendedDataModelingAPI(DataModelingAPI):
+            auto_create_end_nodes (bool): Whether to create missing end nodes for edges when ingesting. By default, the end node of an edge must exist before it can be ingested.
+            auto_create_direct_relations (bool): Whether to create missing direct relation targets when ingesting.
+            skip_on_version_conflict (bool): If existingVersion is specified on any of the nodes/edges in the input, the default behaviour is that the entire ingestion will fail when version conflicts occur. If skipOnVersionConflict is set to true, items with version conflicts will be skipped instead. If no version is specified for nodes/edges, it will do the writing directly.
+            replace (bool): How do we behave when a property value exists? Do we replace all matching and existing values with the supplied values (true)? Or should we merge in new values for properties together with the existing values (false)? Note: This setting applies for all nodes or edges specified in the ingestion call.
+
+        Returns:
+            InstancesApplyResult: Created instance(s)
+
+        """
+        parameters = {
+            "autoCreateStartNodes": auto_create_start_nodes,
+            "autoCreateEndNodes": auto_create_end_nodes,
+            "autoCreateDirectRelations": auto_create_direct_relations,
+            "skipOnVersionConflict": skip_on_version_conflict,
+            "replace": replace,
+        }
+        tasks = [
+            (self._RESOURCE_PATH, task_items, ToolkitRetryTracker(self._http_client_with_retry.config))
+            for task_items in self._prepare_item_chunks(items, self._CREATE_LIMIT, parameters)
+        ]
+        summary = execute_tasks(
+            self._post_with_item_reduction_retry,
+            tasks,
+            max_workers=min(self._config.max_workers, DATA_MODELING_MAX_WRITE_WORKERS),
+            executor=ToolkitConcurrencySettings.get_data_modeling_write_executor(),
+        )
+
+        def unwrap_element(el: T) -> InstanceApply | T:
+            if isinstance(el, dict):
+                return type(items[0])._load(el, cognite_client=self._cognite_client)
+            else:
+                return el
+
+        def str_format_element(el: T) -> str | T:
+            if isinstance(el, InstanceApply):
+                return f"{el.space}:{el.external_id}"
+            return el
+
+        summary.raise_compound_exception_if_failed_tasks(
+            task_unwrap_fn=lambda task: task[1]["items"],
+            task_list_element_unwrap_fn=unwrap_element,
+            str_format_element_fn=str_format_element,
+        )
+        created_resources = summary.joined_results(lambda res: res.json()["items"])
+
+        return InstancesApplyResultList._load(created_resources)
+
+    def _post_with_item_reduction_retry(
+        self,
+        url_path: str,
+        json_payload: dict[str, Any],
```

Comment:
use a proper json typealias here

Reply:
I think that is fine for the moment, hopefully this can be improved in the sdk at some point